src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php

```diff
@@ -25,11 +25,15 @@ final class DateTimeNormalizer implements NormalizerInterface, DenormalizerInter
     public const FORMAT_KEY = 'datetime_format';
     public const TIMEZONE_KEY = 'datetime_timezone';
     public const CAST_KEY = 'datetime_cast';
+    /* if set to true during denormalization DateTime(Immutable) objects will always receive the timezone set in the context
+       and the tz from date strings and timestamps will be ignored */
+    public const PRESERVE_CONTEXT_TIMEZONE_KEY = 'datetime_preserve_context_timezone';
```

Comment:
The `DateTimeNormalizerContextBuilder` and `DateTimeNormalizerContextBuilderTest` must be updated as well.